remote-config/src/main/kotlin/com/rakuten/tech/mobile/remoteconfig/api/ConfigApiClient.kt

```diff
@@ -47,20 +53,53 @@ internal class ConfigApiClient @VisibleForTesting constructor(
         throw InvalidRemoteConfigBaseUrlException(exception)
     }
 
-    fun fetchPath(path: String): Response {
-        val url = requestUrl.newBuilder()
-            .addPathSegments(path)
-            .build()
+    fun fetchPath(path: String, isConfig: Boolean): Response {
+        val builder = requestUrl.newBuilder().addPathSegments(path)
 
-        return client.newCall(
-            Request.Builder()
-                .url(url)
+        if (isConfig) {
```

Comment:
I think it makes more sense to put these conditional params in the `ConfigFetcher` class instead. And then instead of passing the `isConfig` flag, you can pass a nullable `Map` of `params` to `ConfigApiClient#fetchPath` and attach the query params here. That will be more generic in the case we need to add parameters to other requests in the future. What do you think?

Reply:
You're right. It is better to just pass a map of parans. It is more generic.
Updated implementation.